packages/engine/errors/2327.md

```diff
@@ -0,0 +1,21 @@
+---
+original: "Property '{0}' is optional in type '{1}' but required in type '{2}'."
+excerpt: "Property '{0}' in type '{2}' must exist."
+---
+
+Types `{1}` and `{2}` have the same property `{0}` but the property `{0}` is prefixed with `?` in type `{1}` which means it may not exist.
+
+For example, the interface `{1}` declares that there is a property `{0}` which can be a `string` or it should not exist in the object:
+
+```ts
+interface {1} {
```

Comment:
I think that `{1}` can be represented by things other than a name - i.e. an object literal.

Reply:
@mattpocock hm not in this case. The error is explicit in that it'd include an interface name:
> (...) the **interface `{1}`** declares (...)